### PR TITLE
Run Windows NPM kitchen tests on EC2 instead of Azure

### PR DESCRIPTION
**Pull request by derekwbrown**
### What does this PR do?

Run NPM kitchen tests on EC2 instead of Azure

### Motivation

There is a problem with the NPM tests on Azure that doesn't manifest anywhere else.
While we determine the root cause, the tests had been marked "allowed to fail".  Since
there is some probability this is an azure environment thing rather than a code thing,
move the tests so we don't lose the automated testing benefits.

### Possible Drawbacks / Trade-offs

Kitchen tests take longer in EC2

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.

### CODE REVIEW

File: test/kitchen/drivers/ec2-driver.yml [hunk 1]
```diff
@@ -15,20 +15,20 @@ provisioner:
...
   name: ec2
   aws_ssh_key_id: <%= ENV['KITCHEN_EC2_SSH_ID'] %>
-  security_group_ids: <%= [ENV['KITCHEN_EC2_SG_IDS']] || ["sg-7fedd80a","sg-46506837"] %>
+  security_group_ids: <%= ENV['KITCHEN_EC2_SG_IDS'] || ["sg-7fedd80a","sg-46506837"] %>
   region: <%= ENV['KITCHEN_EC2_REGION'] ||= "us-east-1" %>
   instance_type: <%= ENV['KITCHEN_EC2_INSTANCE_TYPE'] ||= 't3.xlarge' %>
   associate_public_ip: false
   subnet_id: <%= ENV['KITCHEN_EC2_SUBNET'] ||= 'subnet-b89e00e2' %>
-  iam_profile_name: <%= ENV['KITCHEN_EC2_IAM_PROFILE_NAME'] ||= nil %>
+  iam_profile_name: <%= ENV['KITCHEN_EC2_IAM_PROFILE_NAME'] ||= '' %>
```
Comment (brycekahle) [inline test/kitchen/drivers/ec2-driver.yml:23]: `nil` is the default value according to the driver docs. Why change to `''`?

File: test/kitchen/drivers/ec2-driver.yml [hunk 2]
```diff
@@ -15,20 +15,20 @@ provisioner:
 driver:
   name: ec2
   aws_ssh_key_id: <%= ENV['KITCHEN_EC2_SSH_ID'] %>
-  security_group_ids: <%= [ENV['KITCHEN_EC2_SG_IDS']] || ["sg-7fedd80a","sg-46506837"] %>
+  security_group_ids: <%= ENV['KITCHEN_EC2_SG_IDS'] || ["sg-7fedd80a","sg-46506837"] %>
```
Comment (brycekahle) [inline test/kitchen/drivers/ec2-driver.yml:18]: This seems like a breaking change for the default value from here: https://github.com/DataDog/datadog-agent/blob/8bb415c33dbb9e6d71c5db8ccd8a6d26c488f3b9/.gitlab/kitchen_common/testing.yml#L153

File: test/kitchen/drivers/ec2-driver.yml [hunk 3]
```diff
@@ -15,20 +15,20 @@ provisioner:
...
   associate_public_ip: false
   subnet_id: <%= ENV['KITCHEN_EC2_SUBNET'] ||= 'subnet-b89e00e2' %>
-  iam_profile_name: <%= ENV['KITCHEN_EC2_IAM_PROFILE_NAME'] ||= nil %>
+  iam_profile_name: <%= ENV['KITCHEN_EC2_IAM_PROFILE_NAME'] ||= '' %>
   spot_price: <%= ENV['KITCHEN_EC2_SPOT_PRICE'] %>
   block_duration_minutes: <%= ENV['KITCHEN_EC2_SPOT_DURATION'] ||= '60' %>
   tags:
     Name: ci-datadog-agent-kitchen
-    Team: agent-platform
+    team: agent-platform
```
Comment (brycekahle) [inline test/kitchen/drivers/ec2-driver.yml:28]: Why the change in case?

File: test/kitchen/drivers/ec2-driver.yml (hunk 1 above)
Comment (derekwbrown) [reply test/kitchen/drivers/ec2-driver.yml:23]: Because when I was running it w/o a profile name, that didn't work (created a parse error), and replacing it with `''` did.

File: test/kitchen/drivers/ec2-driver.yml (hunk 2 above)
Comment (derekwbrown) [reply test/kitchen/drivers/ec2-driver.yml:18]: Note the default value didn't change.  The format of the list changed.  The defaults are set up to run locally from sandbox.  This way, it parses properly when no variable is set.
The link you put above overrides with the correct SG for the environment when run inthe pipeline.

File: test/kitchen/drivers/ec2-driver.yml (hunk 3 above)
Comment (derekwbrown) [reply test/kitchen/drivers/ec2-driver.yml:28]: When I was searching the ec2 console, the other tags seemed to be lower case, so it made it easier to search.  Unfortunately, it's case sensitive.  If it's supposed to be upper case, then I can change it back.

File: .gitlab/kitchen_testing/windows.yml [hunk 4]
```diff
@@ -24,6 +21,20 @@
   # TODO: understand why they fail more often than Linux jobs on network errors.
   retry: 2
 
+.kitchen_os_windows:
+  variables:
+    KITCHEN_OSVERS: "win2008r2,win2012,win2012r2,win2016,win2019,win2019cn,win2022"
```
Comment (iglendd) [inline .gitlab/kitchen_testing/windows.yml:26]: What is win2019cn?

**Comment by iglendd** [comment]
@derekwbrown FYI, I have seen recent PR for Serverless size. You may need to merge/rebase to address the failure above.

File: .gitlab/kitchen_testing/windows.yml (hunk 4 above)
Comment (derekwbrown) [reply .gitlab/kitchen_testing/windows.yml:26]: Chinese.  It tests the widechar support

Comment (iglendd) [reply .gitlab/kitchen_testing/windows.yml:26]: 👍

File: .gitlab/kitchen_testing/windows.yml [hunk 5]
```diff
@@ -109,26 +166,49 @@
...
+    - .kitchen_agent_a7
+    - .kitchen_os_windows_ec2
+  needs: ["deploy_windows_testing-a7"]
+
 # Kitchen: final test matrix (test types * scenarios)
 # ----------------------------------------------
 
-kitchen_windows_installer_npm_install_scenarios-a7:
+##
+## temporarily remove; being run on EC2 instead until Azure issue ironed out
```
Comment (albertvaka) [inline .gitlab/kitchen_testing/windows.yml:179]: Can you specify the issue here? So we can check if it's fixed and move this back to Azure. If this doesn't have to ever be moved back to Azure, then remove the commented code.

File: test/kitchen/drivers/ec2-driver.yml (hunk 2 above)
Comment (albertvaka) [reply test/kitchen/drivers/ec2-driver.yml:18]: If the env var is not set, this evaluates to `[nil] || ["defaultvalue"]`, which resolves to `[nil]`. I think the default will never be used.

File: test/kitchen/drivers/ec2-driver.yml (hunk 3 above)
Comment (albertvaka) [reply test/kitchen/drivers/ec2-driver.yml:28]: Now there's a mix of upper and lower-cased tags. Chose one but make them all the same 😂

**Comment by KSerrania** [comment]
Now that #11847 is merged and NPM driver tests are fixed, should we close this PR?

**Comment by derekwbrown** [comment]
PR no longer necessary.
